planet/Objects.ocd/Animals.ocd/Bat.ocd/Script.c

```diff
@@ -61,155 +61,169 @@ private func Place(int amount, proplist rectangle, proplist settings)
 
 /*-- Behavior --*/
 
-private func FxCoreBehaviorStart(object target, proplist effect, int temp)
-{
-	if (temp)
-		return FX_OK;
-	effect.move_direction = [0, 0];
-	effect.attack_target = nil;
-	effect.last_cave = [nil, nil];
-	effect.time_since_startle = 0;
-	effect.is_expat = false;
-	return FX_OK;
-}
-
-// This time is called every frame for some basic actions.
-private func FxCoreBehaviorTimer(object target, proplist effect, int time)
-{
-	// If the bat is hanging, either start flying or do nothing.
-	if (GetAction() == "Hang")
-	{
-		// Start flying if there is nothing to hang on to.
-		if (!GBackSolid(0, -12))
-			Fly();
-		// Start flying if it is day and not hanging in a tunnel, the bat needs to find a tunnel.
-		if (GetMaterial(0, 0) != Material("Tunnel") && Time->IsDay())
-			Fly();
-		// Start flying by random.
-		if (!Random(150))
-			Fly();		
-		return FX_OK;
-	}
+local CoreBehavior = new Effect {
+    Start = func(int temp) {
+        if(temp) return FX_OK;
+        this.move_direction = [0, 0];
+        this.attack_target = nil;
+        this.last_cave = [nil, nil];
+        this.time_since_startle = 0;
+        this.is_expat = false;
+        return FX_OK;
+    },
+    
+    // This time is called every frame for some basic actions.
+    Timer = func(int time) {
+        // If the bat is hanging, either start flying or do nothing.
+        if(this.Target->GetAction() == "Hang")
```

Comment:
I think I'd prefer to leave out the `this.` and just write `Target`. Does anyone else have an opinion on this?